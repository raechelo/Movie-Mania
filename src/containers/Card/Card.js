import React, { Component } from 'react';
import { connect } from 'react-redux';
import star from '../../media/images/star-solid.svg';
import { toggleFavoriteMovie } from '../../actions';

export class Card extends Component {
  constructor() {
    super()
    this.state = {
      expanded: false
    }
  }

  handleClick = () => {
    this.setState( { expanded: !this.state.expanded } )
  }

  render() {
    const { title, rating, synopsis, posterImage, id, favorited } = this.props.m;
    const expandedCard = 
    <article onClick={this.handleClick}>
      <div className="movie-info">
        <h2>{title}</h2>
        <h6>{rating}</h6>
        <button><i class="fas fa-heart"></i></button>
        <p>{synopsis}</p>
      </div>
      <img src={posterImage} alt={title + ' poster'} />
    </article>

    const heartClasses = favorited ? 'fas fa-heart fa-2x contracted favorited' : 'fas fa-heart fa-2x contracted';
    const contractedCard = <img src={posterImage} onClick={this.handleClick} alt={title + ' poster'} />

    return(
      <div className="Card">
        {!this.state.expanded && <i onClick={() => this.props.toggleFavoriteMovie(id)} className={heartClasses}></i>}
        { this.state.expanded ? expandedCard : contractedCard }
      </div>
    )
  }
}

export const mapDispatchToProps = (dispatch) => ({
  toggleFavoriteMovie: (id) => dispatch(toggleFavoriteMovie(id))
})

export default connect(null, mapDispatchToProps)(Card);