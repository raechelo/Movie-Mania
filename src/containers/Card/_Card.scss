.Card {
  color: white;
  display: flex;
  flex-wrap: wrap;
  justify-content: space-between;
  margin-bottom: 2rem;

  article {
    margin-bottom: 1rem;

    &:hover {
      box-shadow: 2px 2px 2px solid rgb(20, 20, 20);
    }
  }
}