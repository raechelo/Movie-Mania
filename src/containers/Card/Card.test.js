import React from 'react';
import { shallow, mount } from 'enzyme';
import { Card } from './Card';

describe('Card', () => {

  let wrapper;
  let movie = { title: 'IT', rating: 7.9, synopsis: 'The Losers Club attemps to defeat a shapeshifting monster', favorite: false};
  let user = {}
  let mockHandleFavoriteClick = jest.fn()

  beforeEach(() => {
    wrapper = shallow( <Card m={movie} u={user} /> )
  });

  it('should match the snapshot with all the data passed in', () => {
    expect(wrapper).toMatchSnapshot()
  });

  it.skip('should tell a user to log in if they try to favorite without signing in', () => {
    let result = mockHandleFavoriteClick();

    expect(result).toEqual('Login or Signup to favorite movies');
  });

  it('should allow a user to favorite an item if they are logged in', () => {

  });
});