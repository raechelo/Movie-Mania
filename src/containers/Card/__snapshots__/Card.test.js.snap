// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match the snapshot with all the data passed in 1`] = `
<div
  className="Card"
>
  <i
    className="fas fa-heart fa-2x"
    onClick={[Function]}
  />
  <Link
    to="/movies/undefined"
  >
    <img
      alt="IT poster"
    />
  </Link>
</div>
`;
